src/makeCalculator.js

```diff
@@ -37,7 +37,35 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    add(number) {
+      this.result += number;
+    },
+    subtract(number) {
+      this.result -= number;
+    },
+    divide(number) {
+      this.result /= number;
+    },
+    multiply(number) {
+      this.result *= number;
+    },
+
+    operate(operation, number) {
+      operation.bind(this, number)();
```

Comment:
It is less clear than simply calling the operation method directly. Additionally, the bind method creates a new function object, which could lead to performance issues if called repeatedly